### ISSUE TITLE
UseDataClass conflicts with DataClassShouldBeImmutable

### ISSUE THREAD

**Issue by svick** [arturbosch/detekt#1920]
## Expected Behavior of the rule

It should be possible to write a simple `class` with a mutable property that does not violate the rules `UseDataClass` and `DataClassShouldBeImmutable`. Most likely, this means that `UseDataClass` should not be reported for classes that would violate `DataClassShouldBeImmutable` if converted to `data class`.

## Context

Consider the following code:

```kotlin
class C(
    var strings: List<String>
)
```

This produces:

> UseDataClass - [C]

So I convert it to a `data class`:

```kotlin
data class C(
    var strings: List<String>
)
```

But then I get:

> DataClassShouldBeImmutable - [strings]

I don't see a way out of this cycle, so I had to resort to using `@Suppress`.

**Comment by schalkms** [comment]
Thanks for reporting this!
This should be resolved by only proposing classes with immutable member variables as data classes.

**Comment by schalkms** [comment]
Looking at this once more, I'm not sure if detekt should have a different behavior.
Maybe a config option for excluding classes with mutable properties in `UseDataClass` could do the trick here or we could ignore these classes altogether.
On the one hand, the Kotlin Programming Language [guide](https://kotlinlang.org/docs/reference/data-classes.html) mentions that mutable variables are in fact allowed.

> All primary constructor parameters need to be marked as val or var

On the other hand, it's recommended to use immutable properties in data classes.

**Comment by schalkms** [comment]
Maybe a config option for excluding classes with mutable properties in `UseDataClass` could do the trick here or we could be more conservative here and ignore these classes altogether, which I would favor.
On the one hand, the Kotlin Programming Language [guide](https://kotlinlang.org/docs/reference/data-classes.html) mentions that mutable variables are in fact allowed.

> All primary constructor parameters need to be marked as val or var

On the other hand, it's recommended to use immutable properties in data classes.

**Comment by cortinico** [comment]
Here the suggested fix: #1942
